src/ngraph/pass/algebraic_simplification.cpp

```diff
@@ -543,6 +544,93 @@ static bool simplify_reduction(shared_ptr<Node> n)
     return true;
 }
 
+static bool replace_transpose_with_reshape(shared_ptr<Node> n)
+{
+    bool rc = false;
+    auto transpose = as_type_ptr<op::v1::Transpose>(n);
+
+    if (!transpose)
```

Comment:
this check is probably redundant as this func only gets called if the node is transpose.